src/main/java/ticTacToe/ui/Ui.java

```diff
@@ -74,25 +76,46 @@ private String validMarkType(String mark) {
         return mark;
     }
 
-    private String validatePosition(Grid grid, String gridPosition, String mark, Rows rows, int gridSize) {
-        while (!isInsideValidRange(gridPosition, grid)) {
+    private String validatePosition(Grid grid, String mark, Rows rows, int gridSize, String usersInput) {
+        while (!isInsideValidRange(usersInput, grid)) {
             output.println("Invalid position.");
-            promptForPosition(mark, rows, gridSize);
-            gridPosition = input.nextLine();
+            usersInput = promptForNewInput(grid, mark, rows, gridSize);
         }
-        return gridPosition;
+        return usersInput;
     }
 
     private boolean isInsideValidRange(String gridPosition, Grid grid) {
-        return Integer.parseInt(gridPosition) > 0 && Integer.parseInt(gridPosition) <= grid.getCells().size();
+        Integer positionNumber = Integer.parseInt(gridPosition);
+        return positionNumber > 0 && positionNumber <= grid.getCells().size();
+    }
+
+    private String validInput(Grid grid, String mark, Rows rows, int gridSize, String usersInput) {
+        while (isNotNumber(usersInput)) {
+            output.println("Invalid input: position must be a number.");
+            usersInput = promptForNewInput(grid, mark, rows, gridSize);
+        }
+        return usersInput;
+    }
+
+    private boolean isNotNumber(String usersInput) {
+        int position = -1;
+        try {
+            position = Integer.parseInt(usersInput);
+        } catch (NumberFormatException e) {
```

Comment:
Here we have an empty catch block which looks a bit strange. What do you think should happen if an exception is caught here?